File: mobile/lib/utils/validate_interval.dart
/// Validates a string input for use as an interval value (minutes).
///
/// Returns `null` if [input] represents a valid positive integer (≥ 1).
/// Returns a non-null error message string for any invalid input, including:
///   - empty or whitespace-only strings
///   - non-numeric strings
///   - zero
///   - negative integers
///   - decimal numbers (e.g. "1.5", "2.0")
///
/// This function never performs any Firebase write operations.
String? validateInterval(String input) {
  if (input.trim().isEmpty) {
    return 'Please enter a value';
  }

  // Reject decimals before attempting integer parse
  if (input.contains('.')) {
    return 'Please enter a whole number';
  }

  final parsed = int.tryParse(input.trim());
  if (parsed == null) {
    return 'Please enter a valid number';
  }

  if (parsed < 1) {
    return 'Value must be at least 1';
  }

  return null;
}

File: web/src/utils/validateInterval.ts
// Port of the Flutter interval validator (mobile/lib/utils/validate_interval.dart).
// An interval must be a positive integer (>= 1). Anything else — empty, non-numeric,
// zero, negative, or decimal — is rejected.

/**
 * Validates a raw interval input string.
 * @returns `null` when valid, or a human-readable error message when invalid.
 */
export function validateInterval(input: string): string | null {
  const trimmed = input.trim();
  if (trimmed === '') return 'Введіть значення';
  // Only plain non-negative integer digits are accepted (no sign, decimal, or exponent).
  if (!/^\d+$/.test(trimmed)) return 'Має бути ціле число';
  const value = Number(trimmed);
  if (!Number.isInteger(value) || value < 1) return 'Має бути не менше 1';
  return null;
}

/** Convenience predicate wrapping {@link validateInterval}. */
export function isValidInterval(input: string): boolean {
  return validateInterval(input) === null;
}
